### PR TITLE
perf(entity): redirect Entity#getPassengers to the Arclight bridge

### PR DESCRIPTION
**Pull request by danorris709**
Adds an Inject mixin to redirect the base Minecraft `Entity#getPassengers()` to redirect to the Arclight bridge.

This is a performance optimization as the base minecraft `getPassengers` method makes a new instance of the List every time it returns it which can cause some significant performance hits when there's higher player counts.

[Here ](https://spark.lucko.me/upiK7whT2r) is a spark where the issue was found. \
[One](https://i.gyazo.com/2a2ee2f32563106a3c065856c1ecdce3.png) of the sections where it was an issue (it comes up multiple times in that spark). \
`getPassengers ()Ljava/util/List; func_184188_bt`

### CODE REVIEW

**Comment by Rasgnarok** [comment]
Tested minecraft:boat, pixelmon:blue_berry_wood_boat, pixelmon:pixelmon (entity, land mount, water mount, air mount), pixelmon:orange_chair, minecraft:minecart, minecraft:chest_minecart, pixelmon:accro_bike, no errors or crashes.

File: arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/EntityAccessor.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import net.minecraft.entity.Entity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import java.util.Set;
+
+@Mixin(Entity.class)
+public interface EntityAccessor {
+
+    @Invoker void callGetRecursivePassengers(boolean playersOnly, Set<Entity> passengers);
```
Comment (IzzelAliz) [inline arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/EntityAccessor.java]: I prefer not introducing new interfaces like this but using existing bridge interfaces directly.

File: arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/AbstractHorseEntityMixin_Optimize.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package io.izzel.arclight.common.mixin.optimization.general;
```
Comment (IzzelAliz) [inline arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/AbstractHorseEntityMixin_Optimize.java]: All these optimization mixins should be in the arclight-forge/src/main/java/io/izzel/arclight/impl/mixin/optimization/stream package and all mixins should be added to mixin config(mixins.arclight.impl.forge.optimization.stream.json).

**Review by IzzelAliz — CHANGES_REQUESTED** [review]

File: arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/EntityAccessor.java (hunk 1 above)
Comment (danorris709) [reply arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/EntityAccessor.java]: Ah yeah, my bad! I've updated them to use the bridges and added methods to said bridges where necessary :)

File: arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/AbstractHorseEntityMixin_Optimize.java (hunk 2 above)
Comment (danorris709) [reply arclight-common/src/main/java/io/izzel/arclight/common/mixin/optimization/general/AbstractHorseEntityMixin_Optimize.java]: Updated, thank you :)

File: arclight-common/src/main/java/io/izzel/arclight/common/mixin/core/entity/EntityMixin.java [hunk 3]
```diff
@@ -599,6 +595,14 @@ private static boolean isLevelAtLeast(CompoundNBT tag, int level) {
         return passengers;
     }
 
+    @Override
+    public void bridge$getRecursivePassengers(boolean playersOnly, Set<Entity> passengers) {
+        this.getRecursivePassengers(playersOnly, passengers);
+    }
+
+    @Shadow
+    protected abstract void getRecursivePassengers(boolean playersOnly, Set<Entity> passengers);
```
Comment (IzzelAliz) [inline arclight-common/src/main/java/io/izzel/arclight/common/mixin/core/entity/EntityMixin.java]: shadows should be placed at the beginning of the class, inside the @formatter:off block

Comment (danorris709) [reply arclight-common/src/main/java/io/izzel/arclight/common/mixin/core/entity/EntityMixin.java]: Yep, my bad I didn't notice the methods in there too. Updated!

**Pull request merged**